pkg/providerConfigs/adjustDriver.go

```diff
@@ -386,6 +389,12 @@ func PrepareGitRepositories(
 	}
 
 	// COMMIT
+	// * init gitops-template repo
+	gitopsRepo, err := git.PlainInit(gitopsDir, false)
+	if err != nil {
+		return fmt.Errorf("unable to initialize gitops repository at %s. %s", gitopsDir, err.Error())
```

Comment:
```suggestion
		return fmt.Errorf("unable to initialize gitops repository at %s: %w", gitopsDir, err)
```

Doing `err.Error()` converts the error type to a string. We don't want that because then checking for an error means you have to compare strings to strings. If someone were to change the string just one but (say, uppercasing a letter) then it no longer matches.

Using `%w` means two things: when printing it to the screen it will automatically have the string version done (like if you would've magically called `err.Error()`) and the second is that you can use error comparison by types instead of strings.

Reply:
Thank you for the explanation @patrickdappollonio. It's very interesting to learn about these Go features.